### PR TITLE
Added Sous Vide as a node type.

### PR DESCRIPTION
**Pull request by Toby-Mills**
@andrei-tatar please take a look at this for me. It is not yet working.
I have successfully created the new node, and am able to edit config. However when I deploy a flow with the new node, the device status shows as 'Device Invalid', and it does not show up in my Google Home.
I'm not sure what else I need to do?

### CODE REVIEW

**Comment by andrei-tatar** [comment]
@Toby-Mills sure, I can have a look this evening. Will let you know.

File: src/nodes/nora-sousvide.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            name: { name: config.devicename },
+            traits: ['action.devices.traits.TemperatureControl'],
+            state: {
+                online: true,
+                temperatureSetpointCelsius: 50,
+                temperatureAmbientCelsius: 25,
+            },
+            attributes: {
+                queryOnlyTemperatureControl: config.queryOnlyTemperatureControl,
+                temperatureUnitForUX: config.temperatureUnitForUX,
```
Comment (andrei-tatar) [inline src/nodes/nora-sousvide.ts:25]: Main issue is the name in the config is different. This should fix it:
`temperatureUnitForUX: config.unit,`

File: src/nodes/nora-sousvide.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            type: 'action.devices.types.SOUS_VIDE',
+            name: { name: config.devicename },
+            traits: ['action.devices.traits.TemperatureControl'],
+            state: {
+                online: true,
+                temperatureSetpointCelsius: 50,
+                temperatureAmbientCelsius: 25,
+            },
+            attributes: {
+                queryOnlyTemperatureControl: config.queryOnlyTemperatureControl,
```
Comment (andrei-tatar) [inline src/nodes/nora-sousvide.ts:24]: same, name in the config is different:
`queryOnlyTemperatureControl: config.queryOnly,`

**Review by andrei-tatar — CHANGES_REQUESTED** [review]

**Comment by andrei-tatar** [comment]
@Toby-Mills I managed to have a look. Seems the names in the configuration node are different from the ones used in the actual node. `Device invalid` usually means the device doesn't respect the validation schema. I will work on adding some logs for developers to include the validation errors. In this case it was most probably the unit (being required as `C` or `F` but missing).

**Comment by Toby-Mills** [comment]
Thanks @andrei-tatar that did help a bit - this is the problem of lots of copying & pasting!
I've started adding support for the On/Off trait (which Google says is required for Sous Vide), by following the pattern of nora-sensor for implementing multiple traits on a single device. However I'm now back to it reporting 'Device Invalid'.
If you are able to add some support for logging error messages, that might help a lot!

**Comment by andrei-tatar** [comment]
@Toby-Mills  I pushed a commit that will log trace when `Device invalid`. 
You will get something like this in node-red logs: 
```
21 Dec 21:46:01 - [trace] runtime event: {"id":"runtime-state","payload":{"error":"missing-types","type":"warning","text":"notification.warnings.missing-types","types":["noraf-sousvide"]},"retain":true}
```

You should be able to rebase your branch and have the changes. Make sure you enable `trace` level output in your instance of node-red.

**Comment by Toby-Mills** [comment]
@andrei-tatar that's great! That helped a lot. It was a missing property 'on' for the state of the OnOffDevice. The node no longer reports as being invalid, and there are no logged errors.

However I'm not seeing the device show up in the Google Home app. On the page https://smart-nora.eu/my-nora I can see the correct number of devices listed, but the sous vide simply doesn't appear.
Any guesses? Or do you have access to any other logs that might shed light on the situation?

**Comment by andrei-tatar** [comment]
@Toby-Mills I think there's a bug in `nora-firebase-common`. I just noticed the device type shouldn't have an underscore (`_`) in it. https://github.com/andrei-tatar/nora-firebase-common/pull/5/files

Device type should be `action.devices.types.SOUSVIDE`, not `action.devices.types.SOUS_VIDE`.

https://developers.google.com/assistant/smarthome/guides/sousvide

**Comment by Toby-Mills** [comment]
@andrei-tatar cool, that fixed that problem - well spotted!
Now I get a 'Device Invalid' message, but I think that's just because your backend service is not yet using 1.7.1
It lists the allowable device types as including SOUS_VIDE, so I'm guessing it probably just needs to be republished?

**Comment by andrei-tatar** [comment]
Ah yes, of course. Forgot about that :D
Just updated the package in the API.

**Comment by Toby-Mills** [comment]
Nice! I can see my device displayed in my Google Home, and I am receiving messages in Node Red!
Thanks for your help :)
Now I'll work on the UI, and adding support for the other traits.

**Comment by Toby-Mills** [comment]
@andrei-tatar I think this is good to go now. Please review, and let me know if you prefer anything done differently.

**Review by andrei-tatar — APPROVED** [review]
looks good to me! thanks for the contribution

**Pull request merged**